Shanoir issue#823 Study file deletion

jcomedouteau opened the pull request.

michaelkain reviewed (approved):
Hi Jean-Come, it looks totally fine for me. Good idea to clean it in the backend and harmonize it with the entire study edition process. Thank you and with kind regards, Michael

jcomedouteau commented on shanoir-ng-front/src/app/studies/study/study.component.ts:
```diff
@@ -464,14 +464,6 @@ export class StudyComponent extends EntityComponent<Study> {
     public attachNewFile(event: any) {
         let fileToAdd = event.target.files[0];
         this.protocolFiles.push(event.target.files[0]);
```
Well seen, that's quite weird ! :)

I corrected it !

michaelkain commented:
Hi Jean-Come, I test on my machine. I found, that the data-type in the browser (mime-type) is always pdf for downloading protocol files nevertheless what the file type ending is. Would it be easy to optimise?

michaelkain commented:
Next "issue": the database is cleaned when deleting a file, okay, but the file remains on the disk in the study-1 folder, at least in my named volume in /var/studies-data/study-1. I guess the delete should really work? In case of really big files for challenges this will be very important to come along with the disk space size. Thank you and with kind regards, Michael

michaelkain commented:
Hi Jean-Come, sorry for my late idea on this. Would it be possible to add a dialog, that asks for confirmation of file deletion? I have fear that most users could rapidly click and regret what they deleted. What do you think? Cheers, Michael

michaelkain commented:
especially, if we really delete it from the disk, we should alert on that?

jcomedouteau commented:
Yep, the "are you sure" question is quite interesting. And normally the delete actually works on disk, that's all the point of the PR :'(
I'm on it this morning !

jcomedouteau commented:
I tested badly my PR and forgot to test it in the code, that would have saved us some time !
I added the "are you sure" question, and corrected my code with unit tests to test it correctly :)

michaelkain commented on shanoir-ng-studies/src/main/java/org/shanoir/ng/study/service/StudyServiceImpl.java:
```diff
@@ -184,11 +189,21 @@ public Study update(final Study study) throws EntityNotFoundException, MicroServ
+				if (!study.getProtocolFilePaths().contains(filePath)) {
+					// Delete file
+					String filePathToDelete = getStudyFilePath(studyDb.getId(), filePath);
+					System.err.println("COCUOCUOCUCOCUCUCOUCOCCUO    " + filePathToDelete);
```
Hi Jean-Come, not sure if this big cocuocuocuo is ready for prod :-))))

jcomedouteau replied:
Oupsi ! I'm trying to go too fast again !

michaelkain commented:
by the way, thank you a lot for you modifications, I will test them on my machine :-)

michaelkain commented:
sorry for again being picky, but would you mind correcting the question phrase (space between file + question mark)?
![Bildschirmfoto 2021-02-19 um 14 57 54](https://user-images.githubusercontent.com/4560288/108513469-eff58900-72c2-11eb-8b6c-ade83ba078a4.png)

michaelkain commented:
Great, thank you, now the deletion on the disk works!!!

michaelkain commented:
should we not change it to "Study files", in edit study, instead of "Protocol files"? sorry again for this very late change request...

michaelkain commented:
I found a real issue for the challenge, when I upload a big zip file, the downloaded file becomes a pdf, what is wrong from my point of view
![Bildschirmfoto 2021-02-19 um 15 06 12](https://user-images.githubusercontent.com/4560288/108514453-30093b80-72c4-11eb-8ecf-736f0de554b8.png)
![Bildschirmfoto 2021-02-19 um 15 05 57](https://user-images.githubusercontent.com/4560288/108514466-34355900-72c4-11eb-8857-7611ac24d2a1.png)

jcomedouteau commented:
> I found a real issue for the challenge, when I upload a big zip file, the downloaded file becomes a pdf, what is wrong from my point of view
> ![Bildschirmfoto 2021-02-19 um 15 06 12](https://user-images.githubusercontent.com/4560288/108514453-30093b80-72c4-11eb-8ecf-736f0de554b8.png)
> ![Bildschirmfoto 2021-02-19 um 15 05 57](https://user-images.githubusercontent.com/4560288/108514466-34355900-72c4-11eb-8857-7611ac24d2a1.png)

Yup, that's not normal! I forgot to change the "pdf content" on the API, I'll check if it is the main problem

michaelkain reviewed (approved):
I will test locally again

michaelkain commented:
Great! All works now!
